web_poet/pages.py

```diff
@@ -63,10 +70,98 @@ def __init_subclass__(cls, skip_nonitem_fields: bool = False, **kwargs):
 
     async def to_item(self) -> ItemT:
         """Extract an item from a web page"""
+
+        partial_item = await self._partial_item()
+        if partial_item:
+            return partial_item
+
         return await item_from_fields(
             self, item_cls=self.item_cls, skip_nonitem_fields=self._skip_nonitem_fields
         )
 
+    def _get_select_fields(self) -> Optional[SelectFields]:
+        select_fields = getattr(self, "select_fields", None)
+        if not select_fields:
+            return None
+
+        if not isinstance(select_fields, SelectFields):
+            return None
+
+        return select_fields
+
+    @property
+    def fields_to_extract(self) -> Iterable[str]:
+        """Returns an Iterable of field names which should populate the designated
+        ``item_cls``.
```

Comment:
I'm proposing to expose the `fields_to_extract` property method since it's a handy utility for frameworks like **scrapy-poet** to see which fields to populate.

Reply:
Should we choose a name in line with the corresponding attribute of `@field` to exclude or include fields from `to_item` output? (e.g. `to_item_fields`)